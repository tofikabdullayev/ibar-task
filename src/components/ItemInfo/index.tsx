import React, { useState, ChangeEvent, FormEvent } from 'react';
import { Input, FormGroup, Col, Label, Button } from 'reactstrap';

interface Props {
  fields: [string, string];
  isUser: boolean;
  isEditMode: boolean;
  onSubmit: (name: string, email: string) => void;
  onCancel: () => void;
}

const ItemInfo = ({
  fields,
  isUser,
  isEditMode,
  onSubmit,
  onCancel,
}: Props) => {
  const [field1, setField1] = useState(fields[0]);
  const [field2, setField2] = useState(fields[1]);

  const onCancelEdit = () => {
    setField2(fields[1]);
    setField1(fields[0]);
    onCancel();
  };

  const onSubmitEdit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    onSubmit(field1, field2);
  };
  return (
    <form onSubmit={onSubmitEdit}>
      <FormGroup row>
        {isUser && (
          <Label for="name" sm={2}>
            Name:
          </Label>
        )}
        <Col sm={isUser ? 10 : 12}>
          <Input
            value={field1}
            onChange={(e: ChangeEvent<HTMLInputElement>) =>
              setField1(e.target.value)
            }
            disabled={!isEditMode}
            id="name"
          />
        </Col>
      </FormGroup>
      <FormGroup row>
        {isUser && (
          <Label for="email" sm={2}>
            Email:
          </Label>
        )}

        <Col sm={isUser ? 10 : 12}>
          <Input
            type={isUser ? 'text' : 'textarea'}
            value={field2}
            onChange={(e: ChangeEvent<HTMLInputElement>) =>
              setField2(e.target.value)
            }
            disabled={!isEditMode}
          />
        </Col>
      </FormGroup>
      {isEditMode && (
        <div
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            marginBottom: 10,
          }}
        >
          <Button color="primary" style={{ width: '100%', marginRight: 5 }}>
            Submit
          </Button>
          <Button
            color="secondary"
            style={{ width: '100%', marginLeft: 5 }}
            onClick={onCancelEdit}
          >
            Cancel
          </Button>
        </div>
      )}
    </form>
  );
};

export default ItemInfo;
