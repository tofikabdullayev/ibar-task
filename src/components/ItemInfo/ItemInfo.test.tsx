import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ItemInfo from './index';

configure({ adapter: new Adapter() });
test('renders the component', () => {
  const component = shallow(
    <ItemInfo
      fields={['', '']}
      isEditMode={false}
      isUser={true}
      onCancel={() => {}}
      onSubmit={() => {}}
    />
  );
  expect(component).toMatchSnapshot();
});
