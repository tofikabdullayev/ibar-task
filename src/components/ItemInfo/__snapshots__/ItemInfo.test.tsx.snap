// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<form
  onSubmit={[Function]}
>
  <FormGroup
    row={true}
    tag="div"
  >
    <Label
      for="name"
      sm={2}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Name:
    </Label>
    <Col
      sm={10}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Input
        disabled={true}
        id="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </Col>
  </FormGroup>
  <FormGroup
    row={true}
    tag="div"
  >
    <Label
      for="email"
      sm={2}
      tag="label"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      Email:
    </Label>
    <Col
      sm={10}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ]
      }
    >
      <Input
        disabled={true}
        onChange={[Function]}
        type="text"
        value=""
      />
    </Col>
  </FormGroup>
</form>
`;
